Makefile.am

```diff
@@ -41,12 +41,62 @@ pkgconfig_DATA		= syslog-ng.pc
 
 AM_CPPFLAGS		= -I$(top_srcdir)/lib -I$(top_srcdir)/modules -I$(top_builddir)/lib -I$(top_builddir)/modules
 
-AM_CFLAGS =
+
+# configure script gives us -g and/or -O2, but all warning related settings
+# should be here
+
+AM_CFLAGS = -Wall \
+	-Wshadow \
+	-fno-common
+
 if ENABLE_EXTRA_WARNINGS
+
 AM_CFLAGS += \
-	-Wshadow	\
-	-Wno-initializer-overrides \
-	-Wno-error=deprecated-declarations
+        -Wno-empty-body \
```

Comment:
This patch confuses me a lot about the goal of this PR. Before seeing this, I was thinking the goal is to switch as much warnings to errors, as possible, otherwise we silence the errors. Now this patch tries to works the opposite way: downgrading errors to warnings.

Seeing this patch, silencing all warnings would not be less appealing. Lets discuss about this patch irl